### PR TITLE
Publishing Hystrix metrics into Graphite via CodaHale metrics

### PR DESCRIPTION
**Pull request by nurkiewicz**

### CODE REVIEW

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/MetricsConfiguration.groovy [hunk 1]
```diff
@@ -68,4 +70,11 @@ class MetricsConfiguration {
     MetricRegistry metricRegistry(MetricPathProvider metricPathProvider) {
         return new PathPrependingMetricRegistry(metricPathProvider)
     }
+
+    @Bean
+    public HystrixCodaHaleMetricsPublisher hystrixCodaHaleMetricsPublisher(MetricRegistry metricRegistry) {
```
Comment (marcingrzejszczak) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/MetricsConfiguration.groovy]: public is redundant

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/MetricsConfiguration.groovy [hunk 2]
```diff
@@ -68,4 +70,11 @@ class MetricsConfiguration {
...
+        def publisher = new HystrixCodaHaleMetricsPublisher(metricRegistry)
```
Comment (marcingrzejszczak) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/MetricsConfiguration.groovy]: argh!! That 'def' :P

Comment (szpak) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/MetricsConfiguration.groovy]: It is a three-liner with a very long class name used on the right. I don't know if it is much more readable as:
```
HystrixCodaHaleMetricsPublisher publisher = new HystrixCodaHaleMetricsPublisher(metricRegistry)
HystrixPlugins.instance.registerMetricsPublisher(publisher)
return publisher
```

Comment (marcingrzejszczak) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/MetricsConfiguration.groovy]: At least I know what type is there that should be registered :) But of course I'm not an antidef evangelist ;)

Comment (szpak) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/metrics/config/MetricsConfiguration.groovy]: Of course you are not :).

**Comment by nurkiewicz** [comment]
@szpak Agreed, glad you noticed it. Reset is idempotent here (everything is `null`), but still it seems dangerous. Actually it's only needed for tests, since they fail if trying to register publisher twice. An alternative is to call `reset()` after every test that loads `MetricsConfiguration` - which causes a lot of burden in many tests. I'll try with this approach. If #124 is implemented one day, we can simply move cleanup of Hystrix there.

**Comment by nurkiewicz** [comment]
@szpak, OK, proper cleanup before/after every test seems cumbersome and hard to maintain. What if I only call `Hystrix.rest()` in production code when profile is `test`? Seems like a hack as well :-(

**Comment by mchmielarz** [comment]
Calling `HystrixPlugin.reset()` looks like not so pretty workaround here but if there is no other easy way to do this then we have to live with it (at least for the time being).

**Pull request merged**